src/openrct2/ride/RideConstruction.cpp

```diff
@@ -195,11 +196,21 @@ static void ride_remove_vehicles(Ride* ride)
                 sprite_remove(vehicle);
             }
 
-            ride->vehicles[i] = SPRITE_INDEX_NULL;
+            vehicles[i] = SPRITE_INDEX_NULL;
         }
 
         for (size_t i = 0; i < MAX_STATIONS; i++)
-            ride->stations[i].TrainAtStation = RideStation::NO_TRAIN;
+            stations[i].TrainAtStation = RideStation::NO_TRAIN;
+
+        // Also clean up orphaned vehicles for good measure.
+        for (auto* vehicle : TrainManager::View())
```

Comment:
I somewhat wonder if this may cause issues for heavily hacked park creation.

Reply:
It’s only called when removing all vehicles from a ride, so I’d hazard a guess it’s _especially_ beneficial to such parks.